envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsLoggingModeTest.kt

```diff
@@ -0,0 +1,691 @@
+                    - name: authorized-clients
+                      clients: ["echo3", "source-ip-client"]
+                  outgoing:
+                    dependencies:
+                      - service: "echo"
+                      - service: "echo2"
+        """.trimIndent()
+
+        private val echoConfig = Echo1EnvoyAuthConfig.copy(
+            configOverride = proxySettings(unlistedEndpointsPolicy = "blockAndLog"))
+
+        private val echo2Config = Echo2EnvoyAuthConfig.copy(
+            configOverride = proxySettings(unlistedEndpointsPolicy = "log"))
+
+        // language=yaml
+        private val echo3Config = """
+            node:
+              metadata:
+                proxy_settings:
+                  outgoing:
+                    dependencies:
+                      - service: "echo"
+                      - service: "echo2"
+        """.trimIndent()
+
+        private val echoEnvoy by lazy { envoyContainer1 }
+        private val echo2Envoy by lazy { envoyContainer2 }
+        private lateinit var echo3Envoy: EnvoyContainer
+
+        @Container
+        private val sourceIpClient = ToxiproxyContainer(exposedPortsCount = 2).withNetwork(network)
+        private val sourceIpClientToEchoProxy by lazy { sourceIpClient.createProxyToEnvoyIngress(envoy = echoEnvoy) }
+        private val sourceIpClientToEcho2Proxy by lazy { sourceIpClient.createProxyToEnvoyIngress(envoy = echo2Envoy) }
+
+        private val echoLocalService by lazy { localServiceContainer }
+        private val echo2LocalService by lazy { echoContainer2 }
+
+        @JvmStatic
+        @BeforeAll
+        fun setupTest() {
+            setup(appFactoryForEc1 = { consulPort ->
+                EnvoyControlRunnerTestApp(properties = properties(sourceIpClient.ipAddress()), consulPort = consulPort) },
+                envoys = 2,
+                envoyConfig = echoConfig,
+                secondEnvoyConfig = echo2Config
+            )
+            echo3Envoy = createEnvoyContainerWithEcho3Certificate(configOverride = echo3Config)
+            echo3Envoy.start()
+
+            registerServiceWithEnvoyOnIngress(name = "echo", envoy = echoEnvoy, tags = listOf("mtls:enabled"))
+            registerServiceWithEnvoyOnIngress(name = "echo2", envoy = echo2Envoy, tags = listOf("mtls:enabled"))
+
+            waitForEnvoysInitialized()
+        }
+
+        private fun waitForEnvoysInitialized() {
+            untilAsserted {
+                assertThat(echo3Envoy.admin().isEndpointHealthy("echo", echoEnvoy.ipAddress())).isTrue()
+                assertThat(echo3Envoy.admin().isEndpointHealthy("echo2", echo2Envoy.ipAddress())).isTrue()
+                assertThat(echoEnvoy.admin().isEndpointHealthy("echo2", echo2Envoy.ipAddress())).isTrue()
+                assertThat(echo2Envoy.admin().isEndpointHealthy("echo", echoEnvoy.ipAddress())).isTrue()
+            }
+        }
+    }
+
+    fun `should allow echo3 to access status endpoint over https`() {
```

Comment:
Why only echo is allowed to access status endpoints over https?

Reply:
Do you mean echo3? This test do not say that echo3 is the only allowed client to access it. It simply test if it has the access over https.